### PR TITLE
applications: nrf_desktop: Use imply instead of select for modules

### PR DESCRIPTION
**Pull request by MarekPieta**
Change switches to using imply instead of select for application modules and GATT Services. This makes application configuration more flexible.

Jira: NCSDK-19129

### CODE REVIEW

File: applications/nrf_desktop/doc/ble_state.rst [hunk 1]
```diff
@@ -23,14 +23,19 @@ Module events
...
-Implementation details
-**********************
+Configuration
+*************
 
 nRF Desktop uses the Bluetooth LE state module from :ref:`lib_caf` (CAF).
-See the :ref:`CAF Bluetooth LE state <caf_ble_state>` page for implementation details.
+The :ref:`CONFIG_DESKTOP_BLE_STATE <config_desktop_app_options>` Kconfig option selects :kconfig:option:`CONFIG_CAF_BLE_STATE`.
+
+For more information about Bluetooth configuration in nRF Desktop, see :ref:`nrf_desktop_bluetooth_guide`.
```
Comment (kapi-no) [inline applications/nrf_desktop/doc/ble_state.rst:32]: This kind of documentation improvement (moving from section to section), not directly related to the functional change, should be separated into a different commit (e.g. doc clean-up).

Alternatively, you can separate the functional changes (Kconfig changes) from the documentation changes.

However, we may want to rethink the approach of having the follow-up documentation changes in the separate PRs and commits as it makes it hard to revert the changes (functional and documentation changes are not contained in the single commit).

Comment (kapi-no) [reply applications/nrf_desktop/doc/ble_state.rst:32]: (This comment also applies to other parts of the documentation).

Comment (MarekPieta) [reply applications/nrf_desktop/doc/ble_state.rst:32]: The change introduced new Kconfig options `CONFIG_DESKTOP_caf_module_name` and the related doc update is added together with the code. The small documentation structure alignment actually reflects changes in the code (that's why the changes were added together as a single commit).
If you prefer I can split the changes, but then (as you mentioned) doc change will be separated from code.

Comment (MarekPieta) [reply applications/nrf_desktop/doc/ble_state.rst:32]: I could also simply update a commit message to explicitly inform that it introduces these new Kconfig options.

File: applications/nrf_desktop/doc/power_manager.rst [hunk 2]
```diff
@@ -7,8 +7,9 @@ Power manager module
    :local:
    :depth: 2
 
-Use the power manager module to reduce the energy consumption of battery-powered devices.
+Use the Power manager module to reduce the energy consumption of battery-powered devices.
```
Comment (mkapala-nordic) [inline applications/nrf_desktop/doc/power_manager.rst:10]: Is there a convention how to start module names - with capital or lower letter? 
As i can see, in this section we have it starting from capital letter but in `Configuration` section it is starting from lower letter. 
In `settings_loader.rst:10-11` there is similar case.

File: applications/nrf_desktop/doc/power_manager.rst [hunk 3]
```diff
@@ -20,8 +21,16 @@ Module events
 .. note::
     |nrf_desktop_module_event_note|
 
+Configuration
+*************
+
+nrf Desktop uses the power manager module from :ref:`lib_caf` (CAF).
```
Comment (mkapala-nordic) [inline applications/nrf_desktop/doc/power_manager.rst:27]: `nRF` instead of `nrf` + example of starting module name with lower letter

**Review by mkapala-nordic — CHANGES_REQUESTED** [review]

File: applications/nrf_desktop/doc/power_manager.rst (hunk 2 above)
Comment (MarekPieta) [reply applications/nrf_desktop/doc/power_manager.rst:10]: Changed to lowercase

**Review by kapi-no — APPROVED** [review]

File: applications/nrf_desktop/doc/settings_loader.rst [hunk 4]
```diff
@@ -20,13 +20,27 @@ Module events
...
+Settings backend
+================
+
+nRF Desktop devices by default use non-volatile storage settings backend (:kconfig:option:`CONFIG_SETTINGS_BACKEND_NVS`).
+The storage partition is by default located in the internal FLASH.
+
+Settings load in a separate thread
+==================================
+
+Enabling :kconfig:option:`CONFIG_CAF_SETTINGS_LOADER_USE_THREAD` is recommended for keyboard reference design (nRF52832 Desktop Keyboard).
```
Comment (annwoj) [inline applications/nrf_desktop/doc/settings_loader.rst:39]:
```suggestion
Enabling the :kconfig:option:`CONFIG_CAF_SETTINGS_LOADER_USE_THREAD` option is recommended for keyboard reference design (nRF52832 Desktop Keyboard).
```

File: applications/nrf_desktop/doc/power_manager.rst [hunk 5]
```diff
@@ -20,8 +21,16 @@ Module events
 .. note::
     |nrf_desktop_module_event_note|
 
+Configuration
+*************
+
+nRF Desktop uses the power manager module from :ref:`lib_caf` (CAF).
+The :ref:`CONFIG_DESKTOP_POWER_MANAGER <config_desktop_app_options>` Kconfig option selects :kconfig:option:`CONFIG_CAF_POWER_MANAGER` and aligns the default module configuration to the application needs.
+
+The nRF Desktop by default automatically enables :kconfig:option:`CONFIG_PM_POLICY_CUSTOM`.
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/nrf_desktop/doc/power_manager.rst:30]: New wording seems fine :)

**Review by annwoj — APPROVED** [review]

**Review by mkapala-nordic — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

**Pull request merged**